gamemodes/oresgyn/gamemode/init.lua

```diff
@@ -20,6 +20,29 @@ function GM:Initialize()
     roundWaitForPlayers()
 end
 
+function GM:PlayerInitialSpawn(ply)
+    ply:SetSpectator()
+end
+
+function GM:PlayerSpawn(ply)
+    if ply:IsSpectator() then
+        ply:Spectate(OBS_MODE_ROAMING)
+
+        ply:SetEyeAngles(Angle(90, 0, 0))
+
+        if(IsValid(ply.SpectatorPos)) then
+            ply:SetPos(ply.SpectatorPos)
+        end
+    else
+        ply:SetJumpPower(0)
```

Comment:
This will replace the check in GM:SetupMove to disable jumping.
SetupMove was quite laggy when I tested it.